js/calculatefunc.js

```diff
@@ -0,0 +1,99 @@
+const operate = require('./operate');
+
+const calculate = (input, calcObject) => {
+  let { num1, num2, displayValue, operator } = calcObject;
+
+  switch (input) {
+    case 'reset':
+      return { num1: null, num2: null, displayValue: null, operator: null };
+
+    case 'DEL':
+      if (operator === null && num2 === null) {
+        const deleteNum1 = num1 === null ? null : num1.slice(0, -1);
+        return {
+          num1: deleteNum1,
+          num2,
+          displayValue: deleteNum1,
+          operator,
+        };
+      }
+      if (operator !== null && num1 !== null) {
+        const deleteNum2 = num2 === null ? null : num2.slice(0, -1);
+        return {
+          num1,
+          num2: deleteNum2,
+          displayValue: deleteNum2,
+          operator,
+        };
+      }
+      break;
+
+    case '=':
+      if (num1 !== null && num2 !== null && operator !== null) {
+        const result = operate(operator, parseFloat(num1), parseFloat(num2));
+        return {
+          num1: result.toString(),
+          num2: null,
+          displayValue: result.toString(),
+          operator: null,
+        };
+      }
+      return calcObject;
+
+    default:
+      if (/[0-9]/.test(input)) {
```

Comment:
Let's use a fall-through to indicate 0-9 instead of using a default case. For example
```js
 case '0':
 case '1':
 case '2':
 case '3':
// your rest of 4-9 and code here 
```